collection-journey.js

```diff
@@ -1,219 +1,210 @@
+                  type: "point",
+                  longitude: feature.geometry.coordinates[0],
+                  latitude: feature.geometry.coordinates[1],
+                },
+                symbol: pointSymbol,
+                attributes: feature.properties,
+                popupTemplate: template,
+              });
+              graphicsLayer.add(point);
+            } else if (feature.geometry.type === "LineString") {
+              const line = new Graphic({
+                geometry: {
+                  type: "polyline",
+                  paths: feature.geometry.coordinates.map((coord) => [
+                    coord[0],
+                    coord[1],
+                  ]), // ensure this is in the right format
+                },
+                symbol: lineSymbol,
+                attributes: feature.properties,
+                popupTemplate: template,
+              });
+              graphicsLayer.add(line);
             }
+          });
 
-            // Create the map instance.
-            const map = new Map({
-                basemap: baselayer,
-                ground: "world-elevation",
-                layers: layers
-            });
-
-            // Create the map view instance.
-            const view = new MapView({
-                container: "viewDiv",
-                center: [131.034742, -25.345113],
-                zoom: 3,
-                map: map
-            });
-
-            // Merge all extents of layers and go to the merged extent.
-            const layerQueryPromises = [];
-            for (let i = 0; i < layers.length; i++) {
-                layerQueryPromises.push(layers[i].queryExtent());
+          layers.push(graphicsLayer);
+          layerDataMap[config.datasetsConfig[i].id] =
+            config.datasetsConfig[i].layerContent;
+        }
+      }
+
+      // Create the map instance.
+      const map = new Map({
+        basemap: config.basemap,
+        ground: "world-elevation",
+        layers: layers,
+      });
+
+      // Create the map view instance.
+      const view = new MapView({
+        container: "viewDiv",
+        center: [131.034742, -25.345113],
+        zoom: 3,
+        map: map,
+      });
+
+      //TODO FIX THIS
+      // Merge all extents of layers and go to the merged extent.
+      //   const layerQueryPromises = [];
+      //   for (let i = 0; i < layers.length; i++) {
+      //     layerQueryPromises.push(layers[i].queryExtent());
+      //   }
+      //   promiseUtils.eachAlways(layerQueryPromises).then(function (results) {
+      //     let extent = null;
+      //     for (let i = 0; i < results.length; i++) {
+      //       if (typeof results[i].value !== "undefined") {
+      //         if (extent === null) {
+      //           extent = results[i].value.extent;
+      //         } else {
+      //           extent.union(results[i].value.extent);
+      //         }
+      //       }
+      //     }
+      //     view.goTo(extent);
+      //   });
```

Comment:
Is this an outstanding issue or it's a future enhancement?

Reply:
This handles the initial zoom and center view when map is initialized. because I changed the geojsonlayer to graphic layer for journey map (fix bug of line not hidden because two layer was used previously ). And graphic layer don’t support queryExtent()function so this part need to be rewrite. Should be a easy fix.